theano/__init__.py

```diff
@@ -195,11 +191,11 @@ def sparse_grad(var):
     ret = var.owner.op.__class__(sparse_grad=True)(*var.owner.inputs)
     return ret
 
-# This cannot be done in tensor/__init__.py due to a circular dependency -- randomstreams
-# depends on raw_random which depends on tensor.  As a work-around, we import RandomStreams
-# here and inject an instance in tensor.
-from theano.tensor.randomstreams import RandomStreams
+# This cannot be done in tensor/__init__.py due to a circular
+# dependency -- shared_randomstreams depends on raw_random which
+# depends on tensor.  As a work-around, we import RandomStreams here
+# and inject an instance in tensor.
+from theano.tensor.shared_randomstreams import RandomStreams
```

Comment:
great. @carriepl, I let you update the PR.

On Wed, Mar 11, 2015 at 7:01 PM, abergeron <notifications@github.com> wrote:

> In theano/__init__.py
> <https://github.com/Theano/Theano/pull/2589#discussion_r26264907>:
>
> > @@ -195,11 +191,11 @@ def sparse_grad(var):
> >      ret = var.owner.op.__class__(sparse_grad=True)(*var.owner.inputs)
> >      return ret
> >
> > -# This cannot be done in tensor/__init__.py due to a circular dependency -- randomstreams
> > -# depends on raw_random which depends on tensor.  As a work-around, we import RandomStreams
> > -# here and inject an instance in tensor.
> > -from theano.tensor.randomstreams import RandomStreams
> > +# This cannot be done in tensor/__init__.py due to a circular
> > +# dependency -- shared_randomstreams depends on raw_random which
> > +# depends on tensor.  As a work-around, we import RandomStreams here
> > +# and inject an instance in tensor.
> > +from theano.tensor.shared_randomstreams import RandomStreams
>
> Well if you just want to completely remove the interface, then I have no
> objection. We can work out a replacement later.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/2589/files#r26264907>.
>